core/src/docs/rfcs/5485_options_for_reader_with.md

```diff
@@ -0,0 +1,89 @@
+- Proposal Name: `options_for_reader_with`
+- Start Date: 2024-12-31
+- RFC PR: [apache/opendal#5485](https://github.com/apache/opendal/pull/5485)
+- Tracking Issue: [apache/opendal#5486](https://github.com/apache/opendal/issues/5486)
+
+# Summary
+
+Add `if_modified_since` and `if_unmodified_since` options to OpenDAL's `reader_with` API.
```

Comment:
Oh, I double-checked the documentation for `If-Match` and `Range` and confirmed that my previous comments were incorrect. I used to think that if we were using different ranges for reading, the `If-Match` would need to match against different ETags. However, I now understand that I was mistaken. The ETag in `If-Match` and `If-None-Match` can always represent the ETag of the entire file.

---

So it's safe for us to add `if-xxx` API for `Operator::reader_with`!

Reply:
Yeah, at first, i was a bit confused by your comments. 😃